numpy/core/src/multiarray/multiarraymodule.c

```diff
@@ -2940,6 +2940,18 @@ array_arange(PyObject *NPY_UNUSED(ignored), PyObject *args, PyObject *kws) {
         return NULL;
     }
 
+    if (o_stop == NULL) {
+        if (args == NULL || PyTuple_GET_SIZE(args) == 0){
+            PyErr_SetString(PyExc_TypeError,
+                "arange() requires stop to be specified.");
+            return NULL;
```

Comment:
Sorry, I a bit of a sloppy day for me... I just noticed that this branch is missing the `Py_XDECREF(typecode);`

One reason why I am sloppy about these things though, is that nowadays reference count checkers catch these bugs pretty  well most of the time (Unrelated note for anyone who wonders: No I have not run `pytest-leaks` on master/1.20 after branching, only valgrind which is typically not sensitive enough for this type of leak usually.)

Reply:
Maybe a short note in the docs about how to check this would be good. Is there a way to run a checker only on a single test like `python <some command> test_start_stop_kwarg`?

Reply:
Follow up https://github.com/numpy/numpy/pull/17966